sw/device/silicon_creator/rom/sigverify_keys_spx.c

```diff
@@ -0,0 +1,28 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/rom/sigverify_keys_spx.h"
+
+#include "sw/device/silicon_creator/rom/sigverify_keys.h"
+
+#include "otp_ctrl_regs.h"
+
+rom_error_t sigverify_spx_key_get(uint32_t key_id, lifecycle_state_t lc_state,
+                                  const sigverify_spx_key_t **key) {
+  const sigverify_rom_key_header_t *rom_key = NULL;
+  rom_error_t error = sigverify_key_get(
+      (sigverify_key_get_in_params_t){
+          .key_id = key_id,
+          .lc_state = lc_state,
+          .key_array = (const sigverify_rom_key_header_t *)kSigverifySpxKeys,
+          .otp_offset =
+              OTP_CTRL_PARAM_CREATOR_SW_CFG_SIGVERIFY_SPX_KEY_EN_OFFSET,
+          .key_cnt = kSigverifySpxKeysCnt,
+          .key_size = sizeof(sigverify_rom_spx_key_t),
+          .step = kSigverifySpxKeysStep,
+      },
+      &rom_key);
+  *key = &((const sigverify_rom_spx_key_t *)rom_key)->entry.key;
```

Comment:
I need to take a look at this closer, but asking anyway. Is there a chance `rom_key` may ever be NULL here? And if so, we are not checking before dereferencing because this may result in a shutdown escalation?

Reply:
Yes, this was my intention but your comment made me rethink this strategy. I decided to add a `RETURN_IF_ERROR(error)` before this line to help with debugging. Without this, the shutdown would most likely report an illegal read access and it would be harder to understand what's going on.

P.S. I didn't use `HARDENED_RETURN_IF_ERROR` because dereferencing `NULL` will trigger shutdown.